sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -191,22 +206,30 @@ internal class ViewAnnotationManagerImpl(
         )
       }
       if (viewAnnotation.handleVisibilityAutomatically) {
-        val isVisibleNow = (inflatedView.visibility == View.VISIBLE)
-        if (isVisibleNow == viewAnnotation.visible) {
+        val isAndroidViewVisible = (inflatedView.visibility == View.VISIBLE)
+        if (isAndroidViewVisible && viewAnnotation.isVisible ||
+          !isAndroidViewVisible && viewAnnotation.visibility == ViewAnnotationVisibility.INVISIBLE
```

Comment:
just for more clarity
```suggestion
        if ((isAndroidViewVisible && viewAnnotation.isVisible) ||
          (!isAndroidViewVisible && viewAnnotation.visibility == ViewAnnotationVisibility.INVISIBLE)
```